assets/src/components/mapPage.tsx

```diff
@@ -57,70 +65,298 @@ const ToggleMobileDisplayButton = ({
-  const vehicles: VehicleOrGhost[] | null = useSearchResults(
+  const searchVehicles: VehicleOrGhost[] | null = useSearchResults(
     socket,
     searchPageState.isActive ? searchPageState.query : null
   )
-  const onlyVehicles: Vehicle[] = filterVehicles(vehicles)
-  const [mobileDisplay, setMobileDisplay] = useState(MobileDisplay.List)
-  const [selectedVehicleId, setSelectedVehicleId] = useState<VehicleId | null>(
-    searchPageState.selectedVehicleId ?? null
+
+  return (
+    <>
+      <div className="m-map-page__input">
+        <SearchForm formTitle="Search Map" inputTitle="Search Map Query" />
+        {mobileDisplay}
+      </div>
+
+      <hr />
+
+      <div className="m-search-display">
+        {searchVehicles !== null &&
+        thereIsAnActiveSearch(searchVehicles, searchPageState) ? (
+          <SearchResults
+            vehicles={searchVehicles}
+            selectedVehicleId={selectedVehicleId}
+            onClick={selectVehicle}
+          />
+        ) : (
+          <RecentSearches />
+        )}
+      </div>
+    </>
   )
-  const [searchOpen, setSearchOpen] = useState<boolean>(true)
+}
 
-  const liveVehicle: Vehicle | null = selectedVehicleId
-    ? onlyVehicles.find((v) => v.id === selectedVehicleId) || null
-    : null
-  const [showVehicleCard, setShowVehicleCard] = useState<boolean>(
-    searchPageState.selectedVehicleId ? true : false
+// #region Map Display
+const useCachedVehicleById = (
+  selectedVehicleOrGhost: VehicleOrGhost | null,
+  selectedVehicleId: string | null
+) => {
+  const ref = useRef(
+    (selectedVehicleOrGhost &&
+      isVehicle(selectedVehicleOrGhost) &&
+      selectedVehicleOrGhost) ||
+      null
   )
-  const selectedVehicleShapes = useTripShape(liveVehicle?.tripId || null)
 
-  const onSearchCallback = () => {
-    setSelectedVehicleId(null)
+  if (
+    selectedVehicleId === null ||
+    (selectedVehicleOrGhost && isGhost(selectedVehicleOrGhost))
+  ) {
+    ref.current = null
+  } else if (selectedVehicleOrGhost !== null) {
+    ref.current = selectedVehicleOrGhost
   }
+  return ref.current
+}
+
+const useVehicleForIdCached = (
+  socket: Socket | undefined,
+  selectedVehicleId: string | null
+) => {
+  const selectedVehicleOrGhost =
+    useVehicleForId(socket, selectedVehicleId ?? null) || null
+
+  const selectedVehicleRef = useCachedVehicleById(
+    selectedVehicleOrGhost,
+    selectedVehicleId
+  )
+  return selectedVehicleRef
+}
+
+const RouteVehicles = ({
+  selectedVehicleRoute,
+  selectedVehicleId,
+  onPrimaryVehicleSelect,
+}: {
+  selectedVehicleId: VehicleId | null
+  selectedVehicleRoute: RouteId | null
+  onPrimaryVehicleSelect: (vehicle: Vehicle) => void
+}) => {
+  const { socket } = useSocket()
+  const vehicles = useVehiclesForRoute(socket, selectedVehicleRoute)
+  return (
+    <>
+      {filterVehicles(vehicles).map((vehicle: Vehicle) => (
+        <VehicleMarker
+          key={vehicle.id}
+          vehicle={vehicle}
+          isPrimary={true}
```

Comment:
question: would these vehicles not be primary, similar to the non-selected vehicles in the VPP?

Reply:
I dunno! I can check with design on that, from the mockups I'd seen it seemed better to go with the primary vehicle style.

Reply:
Ah yeah I see what you mean! A separate question in this same area - is there an advantage to creating the markers in the `MapPage` component rather than delegating that to the existing `BaseMap` functionality using the `vehicles` and `selectedVehicleId` props?

Reply:
The biggest issue was the autocenterer used all "primary" vehicles by default, and not wanting to add more props/functionality to the base component to support the map page configuration, and "idiomatic" react seemed to be pushing me in the direction of composition over "inheritance" (adding more props..?) to make the map easier to configure for all the different use cases.

I've actually been thinking that we might want to slim down the props that `BaseMap` takes and provide more functionality via composition instead (and preconfigued wrapper components, like the MiniMap). As well as maybe grouping our markers into `pane`s so that it's easier to look through in the react dev tools.

Reply:
Design said yes to the non selected vehicles not being primary 👍🏻

Reply:
I think I covered everything in this thread, so I'm gonna resolve this, but I'll pay attention to this thread if anyone wants to point anything else out!